server.go

```diff
@@ -544,20 +604,26 @@ func (srv *Server) serveUDP(l *net.UDPConn) error {
 	rtimeout := srv.getReadTimeout()
 	// deadline is not used here
 	for {
-		m, s, err := reader.ReadUDP(l, rtimeout)
 		srv.lock.RLock()
+		// check server is still alive
 		if !srv.started {
 			srv.lock.RUnlock()
 			return nil
 		}
+		// and flag that we are processing another packet
+		srv.dnsWg.Add(1)
 		srv.lock.RUnlock()
+		m, s, err := reader.ReadUDP(l, rtimeout)
 		if err != nil {
 			if netErr, ok := err.(net.Error); ok && netErr.Temporary() {
+				srv.dnsWg.Done() // ack processing the UDP packet
```

Comment:
please remove these superfluous comments. dnsWg.Done() is clear enough by itself

Reply:
ok